PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/core/FunctionUtils.java

```diff
@@ -0,0 +1,375 @@
+            FileUtils.copyFile(src, dest);
+        } else {
+            FileUtils.write(src, defaultContent, Charset.defaultCharset());
+        }
+    }
+
+    public static void prepareStagingFolder(Path stagingFolder, Path hostJson, Path localSettingJson, Module module,
+                                            PsiMethod[] methods) throws AzureExecutionException, IOException {
+        final Map<String, FunctionConfiguration> configMap = generateConfigurations(methods);
+        final Path jarFile = JarUtils.buildJarFileToStagingPath(stagingFolder.toString(), module);
+        final String scriptFilePath = "../" + jarFile.getFileName().toString();
+        configMap.values().forEach(config -> config.setScriptFile(scriptFilePath));
+        for (final Map.Entry<String, FunctionConfiguration> config : configMap.entrySet()) {
+            if (StringUtils.isNotBlank(config.getKey())) {
+                final File functionJsonFile = Paths.get(stagingFolder.toString(), config.getKey(), FUNCTION_JSON)
+                        .toFile();
+                FunctionJsonWriter.writeFunctionJsonFile(functionJsonFile, config.getValue());
+            }
+        }
+
+        copyFilesWithDefaultContent(hostJson, new File(stagingFolder.toFile(), "host.json"), DEFAULT_HOST_JSON);
+        copyFilesWithDefaultContent(localSettingJson, new File(stagingFolder.toFile(), "local.settings.json"), DEFAULT_LOCAL_SETTINGS_JSON);
+
+        final List<File> jarFiles = new ArrayList<>();
+        OrderEnumerator.orderEntries(module).productionOnly().forEachLibrary(lib -> {
+            if (StringUtils.contains(lib.getName(), FUNCTION_JAVA_LIBRARY_ARTIFACT_ID)) {
+                return true;
+            }
+
+            if (lib != null) {
+                for (final VirtualFile virtualFile : lib.getFiles(OrderRootType.CLASSES)) {
+                    final File file = new File(stripExtraCharacters(virtualFile.getPath()));
+                    if (file.exists()) {
+                        jarFiles.add(file);
+                    }
+                }
+            }
+            return true;
+        });
+
+        for (final File file : jarFiles) {
+            FileUtils.copyFileToDirectory(file, new File(stagingFolder.toFile(), "lib"));
+        }
+    }
+
+    public static String getTargetFolder(Module module) {
+        if (module == null) {
+            return StringUtils.EMPTY;
+        }
+        final Project project = module.getProject();
+        final MavenProject mavenProject = MavenProjectsManager.getInstance(project).findProject(module);
+        final String stagingFolderName = mavenProject == null ? project.getName() : mavenProject.getProperties().getProperty("functionAppName");
+        return Paths.get(project.getBasePath(), "target", "azure-functions", stagingFolderName).toString();
+    }
+
+    public static String getFuncPath() throws IOException, InterruptedException {
+        final List<String> outputStrings = executeMultipLineOutput(
```

Comment:
use new version of finder:
https://github.com/microsoft/azure-gradle-plugins/blob/master/azure-functions-gradle-plugin/src/main/java/com/microsoft/azure/plugin/functions/gradle/util/FunctionCliResolver.java

Reply:
Copied